src/components/ui/Input/NumberInput.tsx

```diff
@@ -57,20 +57,28 @@ const defaultExceptionSymbols: IExceptionSymbolsProps = {
   exceptPlusOperator: true,
 };
 
-export const NumberInput: FC<INumberInputProps> = ({
-  className,
-  variant = defaultVariant,
-  onKeyDown = handleExceptionSymbols,
-  exceptionSymbolsProps = defaultExceptionSymbols,
-  ...props
-}) => {
-  return (
-    <Input
-      type="number"
-      variant={variant}
-      className={cn('flex-1', className)}
-      onKeyDown={(e) => onKeyDown(e, exceptionSymbolsProps)}
-      {...props}
-    />
-  );
-};
+export const NumberInput = forwardRef<HTMLInputElement, INumberInputProps>(
+  (
+    {
+      className,
+      variant = defaultVariant,
+      onKeyDown = handleExceptionSymbols,
+      exceptionSymbolsProps = defaultExceptionSymbols,
+      ...props
+    },
+    ref,
```

Comment:
Is it necessary to divide `ref` and other props?

Reply:
Absolutely, because it is not a prop but the second parameter of the `forwardRef()`. That's how to use it.

Reply:
Ahh, sorry, now I understand how to use forwardref.